tool/src/js/type_generation/mod.rs

```diff
@@ -398,14 +398,16 @@ impl<'ctx, 'tcx> TyGenContext<'ctx, 'tcx> {
                     } else {
                         None
                     };
-                method_info
-                    .param_conversions
-                    .push(self.gen_js_to_c_for_type(
+                method_info.param_conversions.push(
+                    self.gen_js_to_c_for_type(
                         &param.ty,
                         param_info.name.clone(),
                         struct_borrow_info.as_ref(),
                         alloc,
-                    ));
+                        None,
```

Comment:
issue: I still don't think this is the best way to structure this

 I don't think this parameter should be necessary. We're manually doing `slice.ptr, slice.size` above, so what we have is that two codepaths calling `gen_js_to_c_for_type`, one which uses `...{}.splat()` to splat it, and one which manually spreads it, and we have this `conversion` parameter to deal with that divergence. 

`gen_js_to_c_for_type` _already_ unconditionally returns a `...struct.toFFI()` for structs, so all of its call sites already expect spread operators. I think it should unconditionally return `...slice.splat()` for slices (with caveats on the arena behavior). The call sites should be adjusted to handle that for slices; that will simplify them since currently this call site special-cases slices and manually spreads it.

Reply:
(all of this is an unfortunate consequence of how slices need to be special snowflakes for borrowing, which makes it easy to accidentally write code tha treats them as more special than they need to be)

Reply:
Yeah, it's probably for the best that we try to aim for near-universal usage of `...splat()`  in `gen_js_to_c`, it's just an absolute pain for freeing. I think I'll be able to make a clean solution though.